controllers/autoneg.go

```diff
@@ -58,19 +58,40 @@ func (e *errNotFound) Error() string {
 // Backend returns a compute.Backend struct specified with a backend group
 // and the embedded AutonegConfig
 func (s AutonegStatus) Backend(name string, port string, group string) compute.Backend {
-	if s.AutonegConfig.BackendServices[port][name].Rate > 0 {
+	cfg := s.AutonegConfig.BackendServices[port][name]
+
+	// Extract initial_capacity setting, if set
+	var capacityScaler float64 = 1
+	if capacity := cfg.InitialCapacity; capacity != nil {
+		// This case should not be possible since validateNewConfig checks
+		// it, but leave the default setting of 100% if capacity is less
+		// than 0 or greater than 100
+		if *capacity >= int32(0) && *capacity <= int32(100) {
+			capacityScaler = float64(*capacity) / 100
+
+			// Special case: GCLB does not accept nonzero scaling factors
+			// less than 10%, so increase these to 10%
+			// See: https://cloud.google.com/load-balancing/docs/backend-service#capacity_scaler
+			if *capacity <= int32(10) {
+				capacityScaler = 0.1
```

Comment:
I'm not sure if the desired behavior is to throw an error during validation, or to select 10% (the minimum that GCLB seems to want) or go with 0% (which would be more conservative, since it prevents the backend from receiving any traffic, rather than "more" than an operator might expect)

In practice, I'd expect that people would really only use 0 and 100% via this setting, and manually configure things otherwise via the UI, API, or Google SDK (gcloud) command

Reply:
It's an interesting question, I'm not sure I'd codify this behavior here. It's still the user's responsibility to set it correctly, maybe it would be enough to point to it in the documentation.

Reply:
Removed the 10% check and added a reference to docs